test/Datadog.Trace.Tests/CorrelationIdentifierTests.cs

```diff
@@ -23,7 +23,7 @@ public void TraceIdSpanId_MatchActiveSpan()
             }
         }
 
-        [Fact]
+        [Fact(Skip = "This test is not compatible with the xUnit integration. Neither TraceId or SpanId are Zero.")]
```

Comment:
Makes sense. Not blocking, but do you have ideas on how we could re-enable this test in the future?

Reply:
It depends, if we are trying to test that the `ActiveScope` closes and revert to the parent we can change the test to do something like this:

```csharp
        [Fact()]
        public void TraceIdSpanId_OutsideActiveSpan()
        {
            var baseSpanId = CorrelationIdentifier.SpanId;
            var baseTraceId = CorrelationIdentifier.TraceId;

            using (var parentScope = Tracer.Instance.StartActive("parent"))
            using (var childScope = Tracer.Instance.StartActive("child"))
            {
                // Do nothing
            }

            Assert.Equal<ulong>(baseSpanId, CorrelationIdentifier.SpanId);
            Assert.Equal<ulong>(baseTraceId, CorrelationIdentifier.TraceId);
        }
```

But if we are testing the nullability of the `ActiveScope`, I guess we can create a new `Tracer.Instance` and then restore it later:

```csharp
        [Fact]
        public void TraceIdSpanId_ZeroOutsideActiveSpan()
        {
            var currentTracer = Tracer.Instance;
            Tracer.Instance = new Tracer();

            using (var parentScope = Tracer.Instance.StartActive("parent"))
            using (var childScope = Tracer.Instance.StartActive("child"))
            {
                // Do nothing
            }

            Assert.Equal<ulong>(0, CorrelationIdentifier.SpanId);
            Assert.Equal<ulong>(0, CorrelationIdentifier.TraceId);
            Tracer.Instance = currentTracer;
        }

```

Reply:
In fact, I think the later is the correct change.

Reply:
The second change seems correct. But is that going to cause issues with multiple test collections running in parallel? I assume they would all use the same static Tracer if they're all in the same AppDomain. Or do you know if is there an AppDomain boundary between test collections in the same assembly?

Reply:
I'm not quite sure of that but, we can move this test to a new test collection with the attribute: `[CollectionDefinition(DisableParallelization = true)]` according to the docs:

>Turn off parallelism for specific Test Collection
  [CollectionDefinition(DisableParallelization = true)]
Default: false
Parallel-capable test collections will be run first (in parallel), followed by parallel-disabled test collections (run sequentially).

With that, I think we can avoid that case.

Reply:
Each test class is in a different test collection by default, so to use `[CollectionDefinition]` we would need to place _all_ test classes that access `Tracer.Instance` into the same collection. Adding this attribute to one class is not enough.

> If we need to indicate that multiple test classes should not be run in parallel against one another, then we place them into the same test collection.

Reply:
> Parallel-capable test collections will be run first (in parallel), followed by parallel-disabled test collections (run sequentially).

For this test I think is enough isn't?, all collections without the attributes will run in parallel, then followed by this collection with this special test where we replace the `Tracer.Instance` to do the zero check. 

What I propose is to create a new class with only this `special` test.

Reply:
Yeah, that may be enough. Honestly I think we should avoid using `Trace.Instance` (or other static state) in unit tests if possible. Maybe we should test this instead (this is probably already done elsewhere):
```
var tracer = new Tracer();

using (var parentScope = tracer.StartActive("parent"))
using (var childScope = tracer.StartActive("child"))
{
    Assert.NotEqual<ulong>(0, tracer.ActiveScope?.Span?.SpanId ?? 0);
    Assert.NotEqual<ulong>(0, tracer.ActiveScope?.Span?.TraceId ?? 0);
}

Assert.Equal<ulong>(0, tracer.ActiveScope?.Span?.SpanId ?? 0);
Assert.Equal<ulong>(0, tracer.ActiveScope?.Span?.TraceId ?? 0);
```

Testing `CorrelationIdentifier.TraceId ` seems redundant since it's only a helper API that simply returns `Tracer.Instance.ActiveScope?.Span?.TraceId ?? 0` (same for `SpanId`).

Reply:
Agree!

Reply:
My only disagreement is that these tests are supposed to test the `CorrelationIdentifier` API as a black-box, since that is how it is consumed for the purposes of custom instrumentation / log correlation. Testing the internals works but it's not really testing the API. If we refactor the test as suggested, it should not be part of this test class but moved into another test class like `TracerTests`

Reply:
I completely agree that we should test `CorrelationIdentifier` if possible. I'm just saying that both `Tracer.Instance` and `CorrelationIdentifier` are static, so it will take some hacks and I'm not sure it's worth the trouble.

Reply:
Ok, let's not block on this. I'll propose some testing later since I care very much about the correlation stuff haha